src/main/java/club/minnced/discord/webhook/send/component/SelectMenu.java

```diff
@@ -0,0 +1,109 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package club.minnced.discord.webhook.send.component;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Menu components that can be placed inside a {@link LayoutComponent}
+ *
+ * @see LayoutComponent#addComponent(ActionComponent)
+ */
+
+public class SelectMenu implements ActionComponent{
+
+	private final String customId;
+	private final List<SelectOption> options;
+	private final String placeholder;
+	private final int minValues;
+	private final int maxValues;
+	private boolean disabled;
+
+	public SelectMenu(@NotNull String customId, @NotNull Collection<SelectOption> options, @Nullable String placeholder, int minValues, int maxValues, boolean disabled) {
+		this.customId = customId;
+		this.options = new ArrayList<>(options);
+		this.placeholder = placeholder;
+		this.minValues = minValues;
+		this.maxValues = maxValues;
+		this.disabled = disabled;
+	}
+
+	/**
+	 * A select menu with the defined id and options
+	 * @param customId the dev-defined id
+	 * @param options the options in the dropdown menu
+	 * @return A select menu with the defined id and options
+	 */
+	public static SelectMenu of(@NotNull String customId, @NotNull SelectOption... options) {
+		return new SelectMenu(customId, Arrays.asList(options), null, 1, 1, false);
+	}
+
+	/**
+	 * A select menu with the defined id, placeholder and options
+	 * @param customId the dev-defined id
+	 * @param placeholder the placeholder text if nothing is selected
+	 * @param options the options in the dropdown menu
+	 * @return A select menu with the defined id, placeholder and options
+	 */
+	public static SelectMenu of(@NotNull String customId, @Nullable String placeholder, @NotNull SelectOption... options) {
+		return new SelectMenu(customId, Arrays.asList(options), placeholder, 1, 1, false);
+	}
+
+	@Nullable
+	@Override
+	public String getCustomId() {
+		return customId;
+	}
+
+	@NotNull
+	@Override
+	public Type getType() {
+		return Type.SELECT_MENU;
+	}
+
+	@Override
+	public void withDisabled(boolean disabled) {
+		this.disabled = disabled;
+	}
+
+	@Override
+	public boolean isDisabled() {
+		return this.disabled;
+	}
+
+	@Override
+	public String toJSONString() {
+		JSONObject json = new JSONObject();
+		json.put("type", this.getType().getId());
+		json.put("custom_id", this.customId);
+		json.put("options", this.options);
+		if (placeholder != null)
+			json.put("placeholder", this.placeholder);
+		json.put("min_values", this.minValues);
+		json.put("max_values", this.maxValues);
```

Comment:
This should not be set to 1/1 by default, since that is not the discord default.